Python template refactor

xLinkOut opened the pull request:
Python script template `crypto_price.py` refactor

- Support for both Python 2 and Python 3
- More readable default parameters
- Used `round` instead of build and eval a format string
- Substitute `is` keyword with `==` operator
- Code cleanup and minor optimizations

Maybe more tests are needed but it seems to work well!
(I have not found a guideline for pr, so I kept all commits made but I could do a squash into something like `chg: python template refactor` if you think it's better)

xLinkOut committed:
chg: code cleanup and optimizations

xLinkOut committed:
fix: don't cast primitive type

xLinkOut committed:
doc: default params

xLinkOut committed:
chg: better default params, but need more tests

xLinkOut committed:
chg: is keyword to == operator, no need to use 'is'

xLinkOut committed:
add: package-lock.json ignore

xLinkOut committed:
chg: round mktcap

xLinkOut committed:
add: python 3 support and python 2 backward compatibility

xLinkOut committed:
add: display both '-' and '+' as 'Change % based on opening', like -3.3% or +12% (or 0%)

chrislennon commented:
Hey @xLinkOut - wanted to loop back to you on this, apologies for not doing so already. The changes look great, my only hesitation to merging it so far has been that I would want to do a bunch of manual testing. I was never wise enough to add any automated tests at the start. 

Additionally, with the recent attention this is getting (with current prices), I've been thinking to do a grander refactor of the ugly JS side of the tool, and some further thoughts on an issue [here](https://github.com/chrislennon/Crypto-Touchbar-App/issues/100#issuecomment-803287084). When I find/make time to do so, I will merge in your contributions before taking it in any further direction for sure. 

Thank you for adding this, I hope it's been working well for you for the months since you sent this in.

xLinkOut commented:
Hi @chrislennon! Don't apologize, I appreciate this answer. I have been using this modified version of the template for about a month and I have not had any problems, but obviously, the invitation is always to do as many tests as possible, especially when there are no automatic ones. I'm glad you enjoyed it anyway!

I had also tried to redesign the frontend of the web app, despite not being a web developer. I didn't complete it, the JS part is very "fragile" to changes, however, this is what I managed to complete: rewrote the Touch Bar, to make it much cleaner, sharper and responsive; fixed - with the precious help of @dianaberna - the problems with the two color-pickers, which now also show the preview of the colors directly in the Touch Bar; lastly, added a caching system with `sessionStorage` for the result of the Coin API, to limit the requests at least in the same browser session.

I would be happy if you take a look at it, could be useful for a possible refactor: you can find it in the [redesign](https://github.com/xLinkOut/Crypto-Touchbar-App/tree/redesign) branch in the fork on my profile.

chrislennon commented:
Fragile would be an understatement 😅
Gosh there is some horrible stuff in there...
Can see why I haven't reworked it in many years. 

I'll take a look at the branch and see what I can do when it comes to rebuilding. Thanks for that!
